File: Daily Challanges/2026-01-29.js
function isDigit(char) {
    return char >= '0' && char <= '9';
}

function separateLettersAndNumbers(str) {
  let result = [str[0]]
  for(let i = 1; i < str.length ; i++){
     if(isDigit(str[i-1]) != isDigit(str[i])){
       result.push('-')
     }
     result.push(str[i])
  }
  return result.join('');
}

const test1 = separateLettersAndNumbers("ABC123");
const test2 = separateLettersAndNumbers("Route66");
const test3 = separateLettersAndNumbers("H3LL0W0RLD");
const test4 = separateLettersAndNumbers("a1b2c3d4");

const check1 = "ABC-123";
const check2 = "Route-66";
const check3 = "H-3-LL-0-W-0-RLD";
const check4 = "a-1-b-2-c-3-d-4";

console.log(`Test1: ${test1 === check1}`);
console.log(`Test2: ${test2 === check2}`);
console.log(`Test3: ${test3 === check3}`);
console.log(`Test4: ${test4 === check4}`);

File: Daily Challanges/2026-01-January/2026-01-29.py
def separate_letters_and_numbers(s):
    result = s[0]
    for i in range(1, len(s)):
        if s[i-1].isdigit() != s[i].isdigit():
            result += "-"
        result += s[i]
    return result

test1 = separate_letters_and_numbers("ABC123")
test2 = separate_letters_and_numbers("Route66")
test3 = separate_letters_and_numbers("H3LL0W0RLD")
test4 = separate_letters_and_numbers("a1b2c3d4")

check1 = "ABC-123"
check2 = "Route-66"
check3 = "H-3-LL-0-W-0-RLD"
check4 = "a-1-b-2-c-3-d-4"

print(f"Test1: {test1 == check1}")
print(f"Test2: {test2 == check2}")
print(f"Test3: {test3 == check3}")
print(f"Test4: {test4 == check4}")